plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/ILayer.java

```diff
@@ -72,59 +70,52 @@
      */
     public static final int WORKING = 5;
 
-    /** Listen to changes on this layer.  Each listener can only be added once*/
-    public void addListener( ILayerListener listener );
+    /** Listen to changes on this layer. Each listener can only be added once */
+    public void addListener(ILayerListener listener);
 
     /** remove LayerListener */
-    public void removeListener( ILayerListener listener );
+    public void removeListener(ILayerListener listener);
 
     /**
      * Resource the user associates with this layer. This is not the same as getGeoResource(Class
      * clazz).
      * <p>
      * This is usually the first GeoResource among "friends"
-     * 
+     *
      * @return Resource the user associates with this layer.
      */
     public IGeoResource getGeoResource();
 
-    /**
-     * Returns the first found geoResource that can resolve to clazz.
-     * 
-     * @param clazz the clazz that the returned georesource can resolve to.
-     * @return the first found geoResource that can resolve to clazz.
-     * @deprecated Please use {@link #findGeoResource(Class)}
-     */
-    public <T> IGeoResource getGeoResource( Class<T> clazz );
-
     /**
      * Locate the first IGeoResource that canResolve the provided resource type.
      * <p>
      * Example implementation:
-     * 
-     * <pre><code>
-     * for( IGeoResource resource : getGeoResources() ) {
+     *
+     * <pre>
+     * <code>
+     * for (IGeoResource resource : getGeoResource()) {
      *     if (resource.canResolve(FeatureSource.class)) {
      *         return resource;
      *     }
      * }
      * return null;
-     * </code></pre>
-     * 
+     * </code>
+     * </pre>
+     *
      * </p>
-     * 
+     *
      * @param clazz class of the resource that the IGeoResource claims it can adapt to.
      * @return true if a IGeoResource exists that canResolve to resourceType.
```

Comment:
the javadoc for retrurn is obviously wrong, should be 'the first found geoResource that can resolve to clazz.'
Are you kind to adress this as well?

Reply:
I just pushed a second commit, is this issue resolved than?